server/ntb/commands/sync_topics.py

```diff
@@ -0,0 +1,402 @@
+
+        logger.info("Starting to synchronise Media Topics")
+        vocabularies_json = self._load_vocabularies_json()
+
+        existing_topics = self._get_existing_topics_from_cv(vocabularies_json)
+        updated_topics = self._get_updated_topics_from_iptc(filename)
+        report = self._generate_report_data(updated_topics, existing_topics)
+        self._generate_report(report, updated_topics)
+        self._update_vocabularies_json(vocabularies_json, updated_topics)
+        self._write_changes_to_vocabularies_json(vocabularies_json, output)
+
+    def _get_vocabularies_json_path(self):
+        return path.join(app.config["ABS_PATH"], "data", "vocabularies.json")
+
+    def _load_vocabularies_json(self):
+        """Loads the MediaTopics json from the local filesystem"""
+
+        try:
+            with open(self._get_vocabularies_json_path(), "rb") as f:
+                return json.load(f)
+        except IOError:
+            logger.exception("Failed to load vocabularies.json file")
+            sys.exit(1)
+
+    def _format_json(self, vocabularies_json, output=None):
+        """Write the json data back to the vocabularies.json file"""
+
+        try:
+            file_path = output or self._get_vocabularies_json_path()
+            with open(file_path, "w") as f:
+                f.write(
+                    json.dumps(
+                        vocabularies_json,
+                        ensure_ascii=False,
+                        indent=4
+                    )
+                )
+                f.write("\r\n")
+        except IOError:
+            logger.exception("Failed to write changes to vocabularies.json file")
+            sys.exit(1)
+
+    def _get_existing_topics_from_cv(self, vocabularies_json):
+        """Returns dictionary of MediaTopics from vocabularies.json"""
+
+        for vocab in vocabularies_json:
+            if vocab.get("_id") == "topics":
+                return {
+                    topic["qcode"]: topic
+                    for topic in vocab.get("items") or []
+                }
+
+        return {}
+
+    def _get_updated_topics_from_iptc(self, filename=None):
+        """Get MediaTopics in CV format from IPTC format"""
+
+        return [
+            self._convert_iptc_to_cv(item)
+            for item in self._download_topics_json(filename)
+        ]
+
+    def _download_topics_json(self, filename=None):
+        """Downloads the MediaTopics json file
+
+        Requests all languages as not all entries contain Norwegian translation.
+        Uses a local file instead if `filename` is defined
+        """
+
+        try:
+            if filename:
+                with open(filename, "rb") as f:
+                    data = json.load(f)
+            else:
+                r = requests.get(RESOURCE_URL, params={"lang": "x-all", "format": "json"})
+                data = r.json()
+        except (socket.gaierror, json.JSONDecodeError, IOError):
```

Comment:
Here too, showing the Exception message can help find the trouble (specially important with 3 types of Exception caught).